### DISCUSSION TITLE
The signature at the red arrow is missing. How can I copy it

### DISCUSSION THREAD

**Discussion by linjuchen** [pymupdf/PyMuPDF#2449] [Q&A]
`
import fitz
import pprint as pp
Ndoc = fitz.open()
Npage = Ndoc.new_page()
doc = fitz.open("2.pdf")
for page in doc: 
    widgets = page.widgets()
    for widget in widgets:
        if widget.field_name == 'S1':
            widget.rect = fitz.Rect(50, 450, 150, 700)
            widget.update()
Npage.show_pdf_page(page.rect,doc,0)
Npage.show_pdf_page(fitz.Rect(0,page.rect[3],page.rect[2],page.rect[3]*2),doc,0)

Npage.add_widget(widget)

Ndoc.save("w.pdf")
`

![图片](https://github.com/pymupdf/PyMuPDF/assets/33847736/efe957a2-d9e6-4d7a-80bb-9201c0d4c157)

 The signature at the red arrow is missing. How can I copy it

**Comment by JorjMcKie** [comment]
This is a "Discussions" item.

**Reply by JorjMcKie** [reply]
You cannot. Widgets and annotations cannot be included with method `Page.show_pdf_page()`, see documentation.

**Reply by JorjMcKie** [reply]
You can try and make a PDF -> PDF conversion on the input PDF (i.e. via `doc.convert_to_pdf()`). This may not work if certain font types are used.
But if it does, then annotations and fields are converted to normal page content.
Then you should see the signature image as a normal image.

**Comment by linjuchen** [comment]
`
import fitz
xps = fitz.open("614.pdf")

print(xps.get_sigflags())
pdfbytes = xps.convert_to_pdf()
pdf = fitz.open("pdf", pdfbytes)
pdf.save("some.pdf")
`

 The seal is missing, and some of the original document's text is also missing

**Reply by JorjMcKie** [reply]
Well, as I said - just an idea.
So the bottom line is that this way doesn't work either.

**Comment by linjuchen** [comment]
xps.get_sigflags()  value is 3

**Comment by JorjMcKie** [comment]
So what is left as an alternative, is making a pixmap of the original page and insert that image.
